mimic-iii/buildmimic/duckdb/import_duckdb.py

```diff
@@ -0,0 +1,378 @@
+sqlglot.dialects.duckdb.DuckDB.Generator.TRANSFORMS[exp.DatetimeDiff] = duckdb_date_diff_sql
+sqlglot.dialects.duckdb.DuckDB.Generator.TRANSFORMS[exp.DateDiff] = duckdb_date_diff_sql
+
+# This may not be strictly necessary because the views work without
+# it IF you `use` the schema first... but making them fully qualified
+# makes them work regardless of the current schema.
+def _duckdb_rewrite_schema(sql: str, schema: str):
+    parsed = sqlglot.parse_one(sql, read=sqlglot.dialects.DuckDB)
+    for table in parsed.find_all(exp.Table):
+        for identifier in table.find_all(exp.Identifier):
+            if identifier.this.lower() in tables_in_schema:
+                sql = sql.replace('"'+identifier.this+'"', schema+'.'+identifier.this.lower())
+    # The below (unfinished) causes problems because some munging of functions
+    # occurs in the output. The above approach is kludgy, but works and limits
+    # the blast radius of potential problems regexping SQL.
+    """
+    def transformer(node):
+        if isinstance(node, exp.Table): #and node.name == "a":
+            for id in node.find_all(exp.Identifier):
+                if id.this.lower() in tables_in_schema:
+                    id.this = schema + '.' + id.this.lower()
+                    #print(id)
+            return node
+        return node
+    transformed_tree = parsed.transform(transformer)
+    sql = transformed_tree.sql(dialect=sqlglot.dialects.DuckDB)
+    """
+    return sql
+
+
+def _make_duckdb_query_bigquery(qname: str, qfile: str, conn, schema: str = None):
+    _multischema_trunc_re = re.compile("\"physionet-data\.mimiciii_\w+\.")
+    
+    #TODO: better answer here? should only hit ccs_dx.sql!
+    _too_many_backslashes_re = re.compile("\\\\([\[\.\]])") 
+
+    with open(qfile, "r") as fp:
+        sql = fp.read()
+        sql = re.sub(_too_many_backslashes_re, '\\$1', sql) 
+        try:
+            sql_list = sqlglot.transpile(sql, read="bigquery", write="duckdb", pretty=True)
+        except Exception as e:
+            print(sql)
+            raise e
+        for st in sql_list:
+            sql = re.sub(_multischema_trunc_re, "\"", st)
+
+            if schema is not None:
+                sql = _duckdb_rewrite_schema(sql, schema)
+
+            conn.execute(f"DROP VIEW IF EXISTS {qname}")
+            try:         
+                conn.execute(f"CREATE VIEW {qname} AS " + sql)
+            except Exception as e:
+                print(sql)
+                #print(repr(sqlglot.parse_one(sql)))
+                raise e
+            print(f"CREATED VIEW {qname}")
+
+        #print()
+
+
+def _make_duckdb_query_duckdb(qname: str, qfile: str, conn, schema: str = None):
+    with open(qfile, "r") as fp:
+        sql = fp.read()
+
+        if schema is not None:
+            sql = _duckdb_rewrite_schema(sql, schema)
+
+        try:         
+            conn.execute(f"CREATE VIEW {qname} AS " + sql)
```

Comment:
missing a drop view if exists -> will be fixed by harmonizing duckdb/bigquery subfunctions